### PR TITLE
Feat/multiple reports resolution

### PR DESCRIPTION
**Pull request by cyp3rius**
## Ticket

https://github.com/VirtusLab/strapi-plugin-comments/issues/145 point no. 3

## Summary

What does this PR do/solve?

Provides ability to resolve multiple abuse reports at once

## Test Plan

1. Add couple reports against the comment
2. Open the resolution panel
3. Mark more than one and click "Resolve selected"

### CODE REVIEW

**Review by ltsNotMike — APPROVED** [review]

File: admin/src/components/ReportsReviewTable/index.tsx [hunk 1]
```diff
@@ -90,6 +91,20 @@ const ReportsReviewTable = ({
     );
   };
 
+  const handleItemSelectionChange = (selection, value) => {
+    if (isArray(selection)) {
+      onSelectionChange(value ? selection : [])
+    } else {
+      onSelectionChange([...selectedItems, selection].filter(_ => !value ? selection !== _ : value));
```
Comment (CodeVoyager) [inline admin/src/components/ReportsReviewTable/index.tsx:98]:
```suggestion
      onSelectionChange([...selectedItems, selection].filter(item => value || selection !== item));
```
That's a bit more readable to me. Not 100% sure it's a better solution. Just food for thought.

File: admin/src/components/ReportsReviewTable/index.tsx [hunk 2]
```diff
@@ -98,35 +113,36 @@ const ReportsReviewTable = ({
...
+      <Table colCount={COL_COUNT} rowCount={items.length}>
         <Thead>
           <Tr>
             <Th>
-              <BaseCheckbox aria-label="Select all entries" />
+              <BaseCheckbox 
+                aria-label={getMessage("page.details.panel.discussion.warnings.reports.dialog.selectAll")} 
+                value={areAllItemsSelected()} 
+                disabled={isEmpty(items.filter((_) => !_.resolved))}
+                onValueChange={(value) => handleItemSelectionChange(items.filter((_) => !_.resolved).map(_ => _.id), value)} />
```
Comment (CodeVoyager) [inline admin/src/components/ReportsReviewTable/index.tsx:145]:
```suggestion
                onValueChange={useCallback((value) => handleItemSelectionChange(unresolved.map(prop("id")), value), [unResolved])} />
```

File: admin/src/components/DiscussionThreadItemReviewAction/index.tsx [hunk 3]
```diff
@@ -79,17 +100,42 @@ const DiscussionThreadItemReviewAction = ({
...
+      });
+      if (!isEmpty(items)) {
+        const updatedItems = storedItems.map((_) => ({
+          ..._,
+          resolved: selectedItems.includes(_.id)  ? true : _.resolved,
+        }));
+        setStoredItems(updatedItems);
+        setSelectedItems([], false);
+        onBlockButtonsStateChange(
+          updatedItems.filter((_) => !_.resolved).length === 0
```
Comment (CodeVoyager) [inline admin/src/components/DiscussionThreadItemReviewAction/index.tsx:118]: As above

File: server/services/admin.ts [hunk 4]
```diff
@@ -403,6 +403,47 @@ export = ({ strapi }: StrapiContext): IServiceAdmin => ({
...
+     */
+
+    const reportsToResolve = await strapi.db
+      .query<CommentReport>(getModelUid("comment-report"))
+      .findMany({
+        where: {
+          id: ids,
+          related: commentId,
+        },
+        populate: ['related']
```
[the remark this answers is not in the document]
Comment (cyp3rius) [reply server/services/admin.ts:432]: yep, it's used on the UI

File: server/services/__tests__/admin.test.ts [hunk 5]
```diff
@@ -0,0 +1,660 @@
...
+const setup = (config = {}, toStore = false, database = {}) => {
+  Object.defineProperty(global, "strapi", {
+    value: require("../../../__mocks__/initSetup")(config, toStore, database),
+    writable: true,
+  });
+};
+
+afterEach(() => {
+  Object.defineProperty(global, "strapi", {});
+});
```
[the remark this answers is not in the document]
Comment (cyp3rius) [reply server/services/__tests__/admin.test.ts:19]: Provided `setupStrapi` and `resetStrapi`

File: admin/src/components/ReportsReviewTable/index.tsx [hunk 6]
```diff
@@ -98,35 +113,36 @@ const ReportsReviewTable = ({
...
-      <Table colCount={COL_COUNT} rowCount={storedItems.length}>
+      <Table colCount={COL_COUNT} rowCount={items.length}>
         <Thead>
           <Tr>
             <Th>
-              <BaseCheckbox aria-label="Select all entries" />
+              <BaseCheckbox 
+                aria-label={getMessage("page.details.panel.discussion.warnings.reports.dialog.selectAll")} 
+                value={areAllItemsSelected()} 
+                disabled={isEmpty(items.filter((_) => !_.resolved))}
```
[the remark this answers is not in the document]
Comment (cyp3rius) [reply admin/src/components/ReportsReviewTable/index.tsx:144]: I've used suggestion with `useCallback`

**Pull request merged**